demo-app/src/tx_verifier.rs

```diff
@@ -48,13 +46,6 @@ pub(crate) trait TxVerifier {
 
         Ok(txs)
     }
-
-    /// Runs stateful checks against a Transaction. This method can modify the storage.
-    fn verify_tx_stateful(
```

Comment:
We will definitely want `check_tx` abstraction for the mempool, and it's probably useful to keep the stateful vs. stateless distinction (since stateful checks need to be re-run during block building, but the stateless checks can be skipped). Let's think about how that interacts with the new hooks

Reply:
Discussed offline